From 09eb6f4d18d80a668fb919f6369e4b4eec0e963f Mon Sep 17 00:00:00 2001
From: Divya Theja <divyathe@codeaurora.org>
Date: Thu, 7 Feb 2019 11:41:51 +0530
Subject: [PATCH] Openssl-1.1.*:Replace deprecated BN_init with BN_new

   * API call BN_init(BIGNUM *) used to inititialze object of
     type BIGNUM is removed. In it's place use BIGNUM* BN_new(void)
     to initialize BIGNUM type objects.

Signed-off-by: Pavankumar Vijapur <pvijapur@codeaurora.org>
---
 verify_boot_signature.c | 28 ++++++++++++++--------------
 1 file changed, 14 insertions(+), 14 deletions(-)

diff --git a/verify_boot_signature.c b/verify_boot_signature.c
index 58b7a19..32446d8 100644
--- a/verify_boot_signature.c
+++ b/verify_boot_signature.c
@@ -171,41 +171,41 @@ static int read_signature(int fd, off64_t offset, BootSignature **bs)
  */
 static int validate_signature_block(const BootSignature *bs, uint64_t length)
 {
-    BIGNUM expected;
-    BIGNUM value;
+    BIGNUM *expected;
+    BIGNUM *value;
     int rc = -1;

     if (!bs) {
         return -1;
     }

-    BN_init(&expected);
-    BN_init(&value);
+    expected=BN_new();
+    value=BN_new();

     /* Confirm that formatVersion matches our supported version */
-    if (!BN_set_word(&expected, FORMAT_VERSION)) {
+    if (!BN_set_word(expected, FORMAT_VERSION)) {
         ERR_print_errors(g_error);
         goto vsb_done;
     }

-    ASN1_INTEGER_to_BN(bs->formatVersion, &value);
+    ASN1_INTEGER_to_BN(bs->formatVersion, value);

-    if (BN_cmp(&expected, &value) != 0) {
+    if (BN_cmp(expected, value) != 0) {
         printf("Unsupported signature version\n");
         goto vsb_done;
     }

-    BN_clear(&expected);
-    BN_clear(&value);
+    BN_clear(expected);
+    BN_clear(value);

     /* Confirm that the length of the image matches with the length in
         the authenticated attributes */
     length = htobe64(length);
-    BN_bin2bn((const unsigned char *) &length, sizeof(length), &expected);
+    BN_bin2bn((const unsigned char *) &length, sizeof(length), expected);

-    ASN1_INTEGER_to_BN(bs->authenticatedAttributes->length, &value);
+    ASN1_INTEGER_to_BN(bs->authenticatedAttributes->length, value);

-    if (BN_cmp(&expected, &value) != 0) {
+    if (BN_cmp(expected, value) != 0) {
         printf("Image length doesn't match signature attributes\n");
         goto vsb_done;
     }
@@ -213,8 +213,8 @@ static int validate_signature_block(const BootSignature *bs, uint64_t length)
     rc = 0;

 vsb_done:
-    BN_free(&expected);
-    BN_free(&value);
+    BN_free(expected);
+    BN_free(value);

     return rc;
 }
--
1.9.1
